src/yaml-support/yaml-util.ts

```diff
@@ -90,24 +95,29 @@ export function equalIgnoreCase(a: string, b: string) {
 //      key1: value1
 //      key2: value2
 //
-export function getPropertyValue(mapNode: YamlMap, key: string, ignoreCase: boolean = false) {
+export function getYamlMappingValue(mapRootNode: YamlMap, key: string,
+                                    ignoreCase: StringComparison = StringComparison.Ordinal): string {
     // TODO, unwrap quotes
-    return mapNode.mappings.find((mapping) => mapping.key &&
+    if (!key) {
+        return undefined;
+    }
+    const keyValueItem = mapRootNode.mappings.find((mapping) => mapping.key &&
         (ignoreCase ? key === mapping.key.raw : equalIgnoreCase(key, mapping.key.raw)));
+    return keyValueItem ? keyValueItem.value.raw : undefined;
 }
 
-// get the value in a javascript object with key(may be case sensitive due to the third parameter)
-function getValue(node, key: string, ignoreCase: boolean = false) {
+// get the string value in a javascript object with key(may be case sensitive due to the third parameter)
+function getStringValue(node, key: string, ignoreCase: StringComparison = StringComparison.Ordinal): string {
     if (!node) {
         return undefined;
     }
     if (node.hasOwnProperty(key)) {
-        return node[key];
+        return <string>node[key];
     }
     if (ignoreCase) {
```

Comment:
This needs to be `if (ignoreCase === StringComparison.OrdinalIgnoreCase)` I think.  (You might want to rename the `ignoreCase` parameter to `comparisonOption` or `caseMode` or something like that.)